runtime/compiler/optimizer/J9TransformUtil.cpp

```diff
@@ -969,6 +997,271 @@ J9::TransformUtil::canFoldStaticFinalField(TR::Compilation *comp, TR::Node* node
    return TR_maybe;
    }
 
+static bool isTakenSideOfAVirtualGuard(TR::Compilation* comp, TR::Block* block)
+   {
+   // First block can never be taken side
+   if (block == comp->getMethodSymbol()->getFirstTreeTop()->getEnclosingBlock())
+      return false;
+
+   for (auto edge = block->getPredecessors().begin(); edge != block->getPredecessors().end(); ++edge)
```

Comment:
for an STL iterator it is better to avoid the repeated call to end. Something like:
`for (auto edge = block->getPredecessors().begin(), end = block->getPredecessors().end(); edge != end; ++edge)`

Reply:
Addressed.